src/v/storage/probe.cc

```diff
@@ -114,4 +115,44 @@ void probe::add_initial_segment(const segment& s) {
 void probe::delete_segment(const segment& s) {
     _partition_bytes -= s.reader().file_size();
 }
+
+void readers_cache_probe::setup_metrics(const model::ntp& ntp) {
```

Comment:
I think we should call `segment_iterator_cache` now that readers are stateful, they are more like iterators than readers (simle) ? Might we worth some renaming here.